### ISSUE TITLE
Traceback information missing from AI log files

### ISSUE THREAD

**Issue by Dilvish-fo** [freeorion/freeorion#1690]
### Environment
Ubuntu Linux 16.04
apparently not necessarily happening in (Fedora?) but I am asking for confirmation of that in light of the more specific information below

### Description
AI error logging explicitly triggered by calls to the error() logger , at least as described in #1688 and even after the partial fix there, will result in traceback information being omitted from the log file even though it is properly sent to the human player in chat.

This is not a *huge* problem in that the AI is not broken by it, and the desired log info does get sent to the message window, but in order to be able to handle issues encountered by playtesters who submit logfiles, the traceback info really should appear in the logfile as well.

### Steps to reproduce
 - reintroduce python interface naming errors by reverting 5681081a
 - test with and without the fix from #1688, recompiling
 - start a game with 1 AI (best only 1 to avoid chat window overflow)
 - check chat window and logs

(Quote from #1688)
> It is not necessary to add the exception e as a separate parameter. The python logging module will fetch the exception information on its own with sys.exc_info(), if exc_info=True is passed as a parameter. 

@LGM-Doyle I think that is not fully accurate; there is still a remaining problem in that the traceback is not properly appearing in the log files in these cases.   Please review the 3 snippets below and doublecheck in just which of these cases your results differ.    I am running Python 2.7.12, how about yourself?  I see that there was a 2.7.13 released a couple months ago, but its release notes do not mention logging.

Here is a representative logfile snippet resulting from the python interface name problem *without* the partial fix of 1688 (note, it looks to me like the traceback here is actually from a second exception encountered while trying to handle the logging for the originally-caught exception that was being logged).   

```
00:53:29.176368 [debug] python : ColonisationAI.py:476 - Outpost Fleets Without Missions: 0
00:53:29.176643 [debug] python : ColonisationAI.py:477 - 
00:53:29.176759 [error] root : FreeOrionAI.py:340 - Exception while trying to calculate_priorities
00:53:29.177636 [error] python : traceback.py:13 - Traceback (most recent call last):
00:53:29.178267 [error] python : traceback.py:13 -   File "/home/FO/programs/fo_recov/default/python/AI/freeorion_tools/_freeorion_tools.py", line 111, in emit
00:53:29.178871 [error] python : traceback.py:13 -     msg = self.format(record)
00:53:29.179472 [error] python : traceback.py:13 -   File "/usr/lib/python2.7/logging/__init__.py", line 734, in format
00:53:29.180106 [error] python : traceback.py:13 -     return fmt.format(record)
00:53:29.180688 [error] python : traceback.py:13 -   File "/usr/lib/python2.7/logging/__init__.py", line 465, in format
00:53:29.181290 [error] python : traceback.py:13 -     record.message = record.getMessage()
00:53:29.181889 [error] python : traceback.py:13 -   File "/usr/lib/python2.7/logging/__init__.py", line 329, in getMessage
00:53:29.182494 [error] python : traceback.py:13 -     msg = msg % self.args
00:53:29.183105 [error] python : traceback.py:13 - TypeError: not all arguments converted during string formatting
00:53:29.183672 [error] python : __init__.py:814 - Logged from file FreeOrionAI.py, line 340
```

Here is the corresponding message window chat snippet *after* the partial fix in 1688 (which does not happen at all without that fix):
```
AI_1: AI_Error: AI script error : FreeOrionAI.py:generateOrders():340  - Exception 'empire' object has no attribute 'planetsWithAvailablePP' while trying to calculate_priorities
Traceback (most recent call last):
  File "/home/FO/programs/fo_recov/default/python/AI/FreeOrionAI.py", line 337, in generateOrders
    action()
  File "/home/FO/programs/fo_recov/default/python/AI/PriorityAI.py", line 42, in calculate_priorities
    ColonisationAI.get_colony_fleets()  # sets foAI.foAIstate.colonisablePlanetIDs and foAI.foAIstate.outpostPlanetIDs and many other values used by other modules
  File "/home/FO/programs/fo_recov/default/python/AI/ColonisationAI.py", line 486, in get_colony_fleets
    available_pp = {tuple(el.key()): el.data() for el in empire.planetsWithAvailablePP}
AttributeError: 'empire' object has no attribute 'planetsWithAvailablePP'
```

Here is the corresponding logfile snippet *after* the partial fix in 1688, with one additional line to make more clear that the traceback is missing from the log, not simply missing from the snippet.   Note that this *does not* include the traceback, even though the chat window snippet above does and even though they are both the result of the same call to error() at FreeOrion.py line 340 (not just resulting from the same line, but from the same call):
```
14:42:49.204347 [debug] python : ColonisationAI.py:476 - Outpost Fleets Without Missions: 0
14:42:49.204629 [debug] python : ColonisationAI.py:477 - 
14:42:49.204727 [error] root : FreeOrionAI.py:340 - Exception 'empire' object has no attribute 'planetsWithAvailablePP' while trying to calculate_priorities
14:42:49.205179 [debug] python : ExplorationAI.py:49 - Exploration system considering following system-distance pairs:
```

**Comment by Dilvish-fo** [comment]
after digging around a bit with PR #1720, I thought I should clraify/emphasize that the issue description above was intended to be specifically regarding AI error logging explicitly triggered by calls to the error() logger of the form
```python
   error(""Blah", exc_info=True)
```
as generally happens in the ```except:``` portion of a ```try:  except: ``` block, i.e. caught exceptions.  

For exceptions that are not reported in that fashion, but instead (may) get caught if they appeared in a function that was decorated with the @chat_on_error wrapper, then I *do* already get some traceback info appearing in the log.   **edit actually, it's looking like uncaught errors get a traceback in the log even if the function they appear in was not decorated with @chat_on_error

**Comment by Vezzra** [comment]
Fixed by #1720.
